### DISCUSSION TITLE
example grid, how to implement cancel and live reorder?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [atlassian/pragmatic-drag-and-drop#109] [General]
I am trying to improve the [Grid Example](https://codesandbox.io/s/3nky8j?module=/example.tsx&file=/example.tsx) to learn more about the library, I am coming from dnd-kit & react-dnd.

The first thing that is not clear to me looking at the grid example is there is a cancellation method? I expect if I drag the element back over the starting position the drag should cancel and the grid should stay the same, also dragging the element outside the dnd zone should cancel everything.

The second question I have how would a Sortable grid example look like with pragmatic dnd? If we take the current grid example and want to turn it into a live sortable grid how would that look like in pragmatic dnd?

**Comment by Raphael-Bahuau** [comment]
Up ⬆

**Comment by hungify** [comment]
Hey @ivanjeremic @Raphael-Bahuau !

https://pragmatic-reorder.netlify.app

Here’s a simpler version: [Repo](https://github.com/hungify/pragmatic-reorder), but there are some flicker UI issues to address.

**Reply by ivanjeremic** [reply]
Thank you for the example. It's a good start, and I will try to identify the issue. However, as it stands, it is not usable. I hope we receive more official information soon on how to implement reliable live sorting.

**Comment by Raphael-Bahuau** [comment]
Well done, there is some issues but it's a good start I will look into your code :D .

**Reply by hungify** [reply]
Thank you, it’s difficult to finish that.
Please check the issue: [166](https://github.com/atlassian/pragmatic-drag-and-drop/issues/166#issuecomment-2551366218)
